tests-config-sample.php

```diff
@@ -1,132 +1,68 @@
-
-	// Language
-	'WPLANG'                  => '',
-
-	// Security Hashes (grab from: https://api.wordpress.org/secret-key/1.1/salt)
-	'AUTH_KEY'                => '%%AUTH_KEY%%',
-	'SECURE_AUTH_KEY'         => '%%SECURE_AUTH_KEY%%',
-	'LOGGED_IN_KEY'           => '%%LOGGED_IN_KEY%%',
-	'NONCE_KEY'               => '%%NONCE_KEY%%',
-	'AUTH_SALT'               => '%%AUTH_SALT%%',
-	'SECURE_AUTH_SALT'        => '%%SECURE_AUTH_SALT%%',
-	'LOGGED_IN_SALT'          => '%%LOGGED_IN_SALT%%',
-	'NONCE_SALT'              => '%%NONCE_SALT%%',
-
-	// Security Directives
-	'DISALLOW_FILE_EDIT'      => true,
-	'DISALLOW_FILE_MODS'      => true,
-	'FORCE_SSL_LOGIN'         => false,
-	'FORCE_SSL_ADMIN'         => false,
-
-	// Performance
-	'WP_CACHE'                => true,
-	'WP_MEMORY_LIMIT'         => '96M',
-	'WP_MAX_MEMORY_LIMIT'     => '256M',
-	'EMPTY_TRASH_DAYS'        => 7,
-	'WP_MEMCACHED_KEY_SALT'   => 'tribe',
 
-	// Debug
-	'WP_DEBUG'                => false,
-	'WP_DEBUG_DISPLAY'        => false,
-	'SAVEQUERIES'             => false,
-	'SCRIPT_DEBUG'            => false,
-	'CONCATENATE_SCRIPTS'     => false,
-	'COMPRESS_SCRIPTS'        => false,
-	'COMPRESS_CSS'            => false,
-
-	// Domain Mapping
-	//'SUNRISE'                 => true,
+/*
+ * React dev
+ */
 
-	// Miscellaneous
-	'WP_POST_REVISIONS'       => true,
-	'WP_DEFAULT_THEME'        => 'core'
-);
+define( 'HMR_DEV', false );
 
-// ==============================================================
-// Use defaults array to define constants where applicable
-// ==============================================================
+/*
+ * Debugging
+ */
 
-foreach ( $config_defaults AS $config_default_key => $config_default_value ) {
-	if ( ! defined( $config_default_key ) )
-		define( $config_default_key, $config_default_value );
-}
+define( 'WP_DEBUG', true );
+define( 'SAVEQUERIES', true );
+define( 'WP_DEBUG_DISPLAY', true );
+define( 'SCRIPT_DEBUG', true );
+define( 'WP_CACHE', true );
 
-// ==============================================================
-// Table prefix
-// Change this if you have multiple installs in the same database
-// ==============================================================
+/*
+ * Define a dedicated cache salt for tests to better isolate them
+ * from object caches from other types of request.
+ *
+ * If you're using WPDb for direct database manipulation, the
+ * cache invalidation mechanism might be compromised and you might
+ * face unexpected results in tests that are hard to debug.
+ *
+ * If you'd like to disable object caching completely during tests:
+ * 1 - Replace the line below with "define( 'WP_CACHE_KEY_SALT', md5( microtime( true ) ) );"
+ * 2 - Call "if ( tribe_is_codeception() ) wp_suspend_cache_addition( true );" on a mu-plugin
+ *
+ * Generally speaking, it's a good idea to run tests with cache enabled
+ * because they run faster and you get to test your cache invalidation logic
+ */
+define( 'WP_CACHE_KEY_SALT', 'tests' );
```

Comment:
Have you fully tested this? I've run into so many problems without generating a uniqueID but perhaps that was an old bug.

Reply:
I think you're right. Object cache makes sense in the context of end-to-end tests that replicate user behavior, such as webdriver. For those, we can run a dedicated object cache salt and test the cache validation logic.

However, when you throw into the mix tests that poke the internals to replicate certain scenarios, such as WPDb direct database manipulation and integration tests, which is not end-to-end, we might end up with an inconsistent object caching state. Perhaps we could disable object cache by default, and enable it for webdriver _if/when_ we'd like to test our caching invalidation logic.

I'd just point that, I'm under the impression that if we do use a unique cache salt for every request, we will end up using a lot of RAM, so we would probably want to call `if ( tribe_is_codeception() ) wp_suspend_cache_addition( true );` on a mu-plugin to prevent that

For now, I've added a unique object cache salt for each request, disabling object cache, in practical terms.